plugins/kubernetes/test/models/kubernetes/resource_test.rb

```diff
@@ -795,6 +791,21 @@ def deployment_stub(replica_count)
           end
         end
 
+        let(:error) { +"Foo is invalid: cannot change spec.bar" }
+
+        before { template[:metadata][:annotations] = {"samson/force_update": "true"} }
+
+        it "re-creates when updating is not possible" do
```

Comment:
this may be dangerous if it is a StatefulSet with PersistentVolume which has `persistentVolumeReclaimPolicy` set to `Delete`.

Reply:
up to the user to not use it then ... I'd prefer not hardcoding more rules